indicate_later is not synchronised with the state of the device
thus inducing bad behavior on RINGOUT and RINGING state.
Index: asterisk-chan-sccp-0~prerelease+svn2187/src/sccp_pbx.c
===================================================================
--- asterisk-chan-sccp-0~prerelease+svn2187.orig/src/sccp_pbx.c	2011-11-03 10:37:23.584110851 -0400
+++ asterisk-chan-sccp-0~prerelease+svn2187/src/sccp_pbx.c	2011-11-03 10:39:02.155362282 -0400
@@ -321,7 +321,10 @@
 			if (linedevice->device->dndFeature.enabled && linedevice->device->dndFeature.status == SCCP_DNDMODE_REJECT)
 				continue;
 
-			sccp_indicate_later(linedevice->device, c, SCCP_CHANNELSTATE_RINGING);
+			sccp_channel_lock(c);
+                        sccp_indicate_locked(linedevice->device, c, SCCP_CHANNELSTATE_RINGING);
+                        sccp_channel_unlock(c);
+
 			isRinging = TRUE;
 			if (c->autoanswer_type) {
 
@@ -913,7 +916,9 @@
 			// Allow signalling of RINGOUT only on outbound calls.
 			// Otherwise, there are some issues with late arrival of ringing
 			// indications on ISDN calls (chan_lcr, chan_dahdi) (-DD).
-			sccp_indicate_later(c->device, c, SCCP_CHANNELSTATE_RINGOUT);
+                        sccp_channel_lock(c);
+                        sccp_indicate_locked(c->device, c, SCCP_CHANNELSTATE_RINGOUT);
+                        sccp_channel_unlock(c);
 		}
 		break;
 	case AST_CONTROL_BUSY:
